docs/advanced-configuration.md

```diff
@@ -226,11 +229,57 @@ The default configurations for the control plane receivers can be found in
 Here are the documentation links that contain configuration options and supported metrics information for each receiver
 used to collect metrics from the control plane.
 * [smartagent/coredns](https://docs.splunk.com/Observability/gdi/coredns/coredns.html)
+* [smartagent/etcd](https://docs.splunk.com/Observability/gdi/etcd/etcd.html)
 * [smartagent/kube-controller-manager](https://docs.splunk.com/Observability/gdi/kube-controller-manager/kube-controller-manager.html)
 * [smartagent/kubernetes-apiserver](https://docs.splunk.com/Observability/gdi/kubernetes-apiserver/kubernetes-apiserver.html)
 * [smartagent/kubernetes-proxy](https://docs.splunk.com/Observability/gdi/kubernetes-proxy/kubernetes-proxy.html)
 * [smartagent/kubernetes-scheduler](https://docs.splunk.com/Observability/gdi/kubernetes-scheduler/kubernetes-scheduler.html)
 
+### Setting up Etcd metrics
+
+The etcd metrics cannot be collected out of box because etcd requires TLS authentication for communication.
+
+The easiest way to set up the TLS authentication for etcd metrics is to retrieve the client certificate and key from an
+etcd pod and directly insert them into the values.yaml. The helm chart will set up the rest. The helm chart will add
+the client certificate and key to the otel-collector agent's kubernetes secret and then configure the etcd receiver to
+use them. The etcd TLS client certificate and key play a critical role in the security of the cluster, handle them with
```

Comment:
> The etcd TLS client certificate and key play a critical role in the security of the cluster, handle them with
care and avoid storing them in unsecured locations.

This being said, having them in plaintext in a values file seems less than ideal as a requirement on our part. Do you know if there's a standard way of reading from a file at installation? Or would it make sense to have them create the secrets themselves and specify their name so we can use them in a mount?

Reply:
I think it's ok to have an option to provide the certificates in values.yaml. We do the same for all other secrets. It's just being used to create k8s secrets. It doesn't need to be persisted. At the same time we should provide an option to use k8s secrets created manually as we do for other secrets in this chart

Reply:
Also etcd certificated must be present on master nodes, right? So I believe users can just mount them to the container if they want to. Can we test that approach and document as an option?

Reply:
Also etcd certificated must be present on master nodes, right? So I believe users can just mount them to the container if they want to. Can we test that approach and document as an option?

Reply:
That's fine. We need to test and document this as an option asking user to install the helm chart 2 times with appropriate nodeSelectors/tolerations

Reply:
We have proposed 3 ways to handle etcd secrets here.
1. Have the user supply the etcd cert values to the helm chart in values.yaml or with -set. Then have the helm chart handle creating the secret and using the secret.
2. Have the users supply a supply a secret name that contains the etcd certs. Then have helm use the supplied secret.
3. Have the user deploy two instances of the helm chart; one for master nodes and one for worker nodes. The helm chart uses the etcd certs that are present on master nodes and doesn't use any kubernetes secrets.

All 3 of these methods sound acceptable to me. Let me know if you think otherwise.

Reply:
Sounds good to me. Thanks

Reply:
Update:
Adding the 3rd method mentioned above is a lot more work than I anticipated. To keep the scope of this ticket smaller, I am only going to support the 1st and 2nd methods mentioned above. We can add the 3rd method in another ticket at a later time.